packages/rs-drive-abci/src/platform_types/snapshot/mod.rs

```diff
@@ -168,4 +207,32 @@ mod tests {
             fs::remove_dir_all(grove_dir.path()).unwrap();
         }
     }
+
+    #[test]
+    fn test_offer_snapshot() {
+        let grove_dir = tempfile::tempdir().unwrap();
+        let replication_dir = tempfile::tempdir().unwrap();
+        let grove = GroveDb::open(grove_dir.path()).unwrap();
+        let mut manager = Manager::new("".to_string(), Some(3), Some(1));
+        let app_hash = vec![1, 2, 3, 4, 5];
+        let snapshot_1000 = abci::Snapshot {
+            height: 1000,
+            version: 0,
+            hash: app_hash.clone(),
+            metadata: vec![],
+        };
+        manager
+            .offer_snapshot(&grove, snapshot_1000.clone(), app_hash.clone())
+            .unwrap();
+        let snapshot_2000 = abci::Snapshot {
+            height: 2000,
+            version: 0,
+            hash: app_hash.clone(),
+            metadata: vec![],
+        };
+        manager
+            .offer_snapshot(&grove, snapshot_2000.clone(), app_hash.clone())
+            .unwrap();
+        fs::remove_dir_all(grove_dir.path()).unwrap();
```

Comment:
The tempdir function creates a directory in the file system and returns a TempDir. The directory will be automatically deleted when the TempDirs destructor is run.